### PR TITLE
feat(cli): serial multi-stack deploys

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
The goal of this PR is to make a first step towards multi-stack deployments by running each deployment after another (with no parallelism). This ships a lot of the important infrastructure (running in the correct order, handling aborts, etc) while leaving the management of multiple deployments at the same time open for a follow up PR.

### CODE REVIEW

**Comment by jsteinich** [comment]
Synth, output, and maybe diff should support multiple stacks.

We should aim for parity with the AWS CDK with how multiple stacks are specified (https://docs.aws.amazon.com/cdk/v2/guide/cli.html#cli-stacks)

**Comment by DanielMSchmidt** [comment]
> Synth, output, and maybe diff should support multiple stacks.

Yeah, we are going to do this in separate PRs for synth and output 👍 Not sure how diff should be working, if there is a dependency there might be no way to diff (if the provider requires a previously provisioned endpoint to be there for example). How do you think this should look like?

> We should aim for parity with the AWS CDK with how multiple stacks are specified

Yeah agreed, will be done in a later PR :)

**Comment by jsteinich** [comment]
> Not sure how diff should be working, if there is a dependency there might be no way to diff (if the provider requires a previously provisioned endpoint to be there for example). How do you think this should look like?

I would expect it to try and diff the stacks using the same dependency ordering as other places. Yes, sometimes the diff would fail, but that's not really any different than how it works in HCL.

File: packages/cdktf-cli/lib/stack-execution.ts [hunk 1]
```diff
@@ -84,41 +79,15 @@ function extractError(_context: any, event: any): string {
...
-      `Found more than one stack, please specify a target stack. Run cdktf ${
-        context.targetAction || "<verb>"
-      } <stack> with one of these stacks: ${stacks
-        .map((s) => s.name)
-        .join(", ")} `
-    );
-  }
-
-  return stacks[0];
+  throw new Error("No stack selected (add some more data here)");
```
[the remark this answers is not in the document]
Comment (DanielMSchmidt) [reply packages/cdktf-cli/lib/stack-execution.ts]: yeah :D

**Pull request merged**